codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/auth/factories/HttpBasicAuthSchemeFactory.java

```diff
@@ -21,35 +21,29 @@
 import software.amazon.smithy.ruby.codegen.config.ClientConfig;
 
 public final class HttpBasicAuthSchemeFactory {
+
+    public static final ClientConfig HTTP_LOGIN_PROVIDER = ClientConfig.builder()
```

Comment:
Maybe there should be factories for all of the providers?

Reply:
I'm not sure its worth creating separate classes for all for all of the client configs since they aren't generally shared.  This was a bit of a special case.  But maybe it should be its own file?

Reply:
It's probably fine as is